docs/filesystems.txt

```diff
@@ -0,0 +1,144 @@
+This ensures the integrity of transactions and backups.
+
+
+Configuration
+-------------
+
+Before using the GridFS driver, you will need to install the Flysystem GridFS package via the Composer package manager:
+
+.. code-block:: bash
+
+   composer require league/flysystem-gridfs
+
+Configure `Laravel File Storage <https://laravel.com/docs/{+laravel-docs-version+}/filesystem>`__,
+to use the ``gridfs`` driver in ``config/filesystems.php``:
+
+.. code-block:: php
+
+   'disks' => [
+       'gridfs' => [
+           'driver' => 'gridfs',
+           'connection' => 'mongodb',
+           'database' => 'files',
+           'bucket' => 'fs',
+           'prefix' => '',
+           'read-only' => false,
+           'throw' => false,
+       ],
+   ],
+
+.. list-table::
+   :header-rows: 1
+   :widths: 25 75
+
+   * - Setting
+     - Description
+
+   * - ``driver``
+     - **Required**. Specifies the filesystem driver to use. Must be ``gridfs`` for MongoDB.
+
+   * - ``connection``
+     - The database connection used to store jobs. It must be a ``mongodb`` connection. The driver uses the default connection if a connection is not specified.
+
+   * - ``database``
+     - Name of the MongoDB database fot the GridFS bucket. Use the database of the connection if not specified.
+
+   * - ``bucket``
+     - Name of the GridFS bucket. A database can contain multiple buckets identified by their name. Defaults to ``fs``.
+
+   * - ``prefix``
+     - Specifies a prefix for the name of the files that are stored in the bucket. Using a distinct bucket is recommended
+       in order to store the files in a different collection.
+
+   * - ``read-only``
+     - If ``true``, writing to the GridFS bucket is disabled. Write operations will return ``false`` or throw exceptions
+       depending on the configuration of ``throw``. Defaults to ``false``.
+
+   * - ``throw``
+     - If ``true``, exceptions are thrown when an operation cannot be performed. Defaults to ``false``, the operations
+       return ``true`` in case of succes, ``false`` in case of error.
+
+If you have specific requirements, you can use a factory or a service name to define the bucket. The options
+``connection`` and ``database`` are ignored when a ``MongoDB\GridFS\Bucket`` is provided:
+
+.. code-block:: php
+
+    use Illuminate\Foundation\Application;
+    use MongoDB\GridFS\Bucket;
+
+   'disks' => [
+       'gridfs' => [
+           'driver' => 'gridfs',
+           'bucket' => static function (Application $app): Bucket {
+               return $app['db']->connection('mongodb')
+                   ->getMongoDB()
+                   ->selectGridFSBucket([
+                       'bucketName' => 'avatars',
+                       'chunkSizeBytes' => 261120,
+                   ]);
+           },
+       ],
+   ],
+
+Usage
+-----
+
+The benefits of using Laravel File Storage facade, is that it provides a common
+interface for all the supported file systems. Use the ``gridfs`` disk in the
```

Comment:
I don't think the comma serves a purpose here. Consider instead:

```suggestion
A benefit of using Laravel's File Storage facade is that it provides a common
interface for all the supported file systems. Use the ``gridfs`` disk in the
```

Alternatively "the Laravel's File Storage". Also, the facade is just the API using `Storage::`, no? Would it be more correct to refer to this as a component? I'll defer to you.

I assume this is what we're referring to: https://laravel.com/docs/11.x/filesystem